src/coreclr/scripts/jitrollingbuild.py

```diff
@@ -172,6 +182,111 @@ def determine_jit_name(coreclr_args):
         raise RuntimeError("Unknown OS.")
 
 
+def process_git_hash_arg(coreclr_args):
+    """ Process the -git_hash argument.
+
+        If the argument is present, use that to download a JIT.
+        If not present, try to find and download a JIT based on the current environment:
+        1. Determine the current directory git hash using:
+             git rev-parse HEAD
+           Call the result `current_git_hash`.
+        2. Determine the baseline: where does this hash meet `main` using:
+             git merge-base `current_git_hash` main
+           Call the result `base_git_hash`.
+        3. Figure out the latest hash, starting with `base_git_hash`, that contains any changes to
+           the src/coreclr/jit directory. (We do this because the JIT rolling build only includes
+           builds for changes to this directory. So, this logic needs to stay in sync with the logic
+           that determines what causes the JIT rolling build to run. E.g., it should also get
+           rebuilt if the JIT-EE interface GUID changes. Alternatively, we can take the entire list
+           of changes, and probe the rolling build drop for all of them.)
+        4. Starting with `base_git_hash`, and possibly walking to older changes, look for matching builds
+           in the JIT rolling build drops.
+        5. If a JIT directory in Azure Storage is found, set coreclr_args.git_hash to that git hash to use
+           for downloading.
+
+    Args:
+        coreclr_args (CoreclrArguments) : parsed args
+
+    Returns:
+        Nothing
+
+        coreclr_args.git_hash is set to the git hash to use
+
+        An exception is thrown if the `-git_hash` argument is unspecified, and we don't find an appropriate
+        JIT to download.
+    """
+
+    if coreclr_args.git_hash is not None:
+        return
+
+    # Do all the remaining commands, including a number of 'git' commands including relative paths,
+    # from the root of the runtime repo.
+
+    with ChangeDir(coreclr_args.runtime_repo_location):
+        command = [ "git", "rev-parse", "HEAD" ]
+        print("Invoking: {}".format(" ".join(command)))
+        proc = subprocess.Popen(command, stdout=subprocess.PIPE)
```

Comment:
we should try to use `run_command` instead. It has pretty much most of the functionality of getting back stdout, return code, etc.

Reply:
The trick with using `run_command` is it has a different output model. We could introduce a `verbose_output` option. I propose to defer changing this.